### ISSUE TITLE
Text/prepend operator misbehaves

### ISSUE THREAD

**Issue by adamwasila** [Jeffail/benthos#254]
I found this peculiar error where such simplest pipeline fails to deliver proper output. 

First build anything that uses `text` processor with `prepend` operator, eg.:

```
input:
  type: benthos_lab
pipeline:
  processors:
    - type: text
      text:
        operator: prepend
        value: "--"
output:
  type: benthos_lab
```

Then prepare input; remember to keep it under 6 bytes (as total lenght with prefix should not exceed 8 bytes), eg.:

```
aaaaaa
bbbb
cc
```

Observe output:

```
Compiled successfully.
--ccbbaa
--ccbb
--cc
```

Where proper output should be:
```
--aaaaaa
--bbbb
--cc
```

(Lab example can be found here: https://lab.benthos.dev/l/6htCxZXE7z2 )

**Comment by adamwasila** [comment]
Worry not, root cause has been found and is patched already (see PR).

Though, if you insist to worry (a bit), I found few other places where similar problems may occur, eg. here:

https://github.com/Jeffail/benthos/blob/master/lib/input/reader/lines.go#L181

Well, first of all, I'm not sure if slice returned by Bytes() should be used after another call to Write() (which will happen for multipart message) as docs clearly forbids it (implementation may be a bit relaxed though). But that's just part of the problem. Real headache will start when someone tries to use such message and uses it as first argument of append. Then, if capacity allows it, append will happen in-place, destroying everything that appears next in the buffer (most probably following message part).

Hopefully this helps somehow to carry on because I can't provide more code or debug time. I have to leave it for now to focus on other tasks, eg. cassandra output I promised to investigate.
